src/whylogs/core/statistics/constraints.py

```diff
@@ -318,23 +352,31 @@ def merge(self, other) -> "SummaryConstraint":
     @staticmethod
     def from_protobuf(msg: SummaryConstraintMsg) -> "SummaryConstraint":
 
-        if msg.HasField("value") and not msg.HasField("second_field") and not msg.HasField("between"):
+        if msg.HasField("reference_set") and not msg.HasField("value") and not msg.HasField("second_field") and not msg.HasField("between"):
+            return SummaryConstraint(
+                msg.first_field,
+                msg.op,
+                reference_set=set(msg.reference_set),
+                name=msg.name,
+                verbose=msg.verbose,
+            )
+        elif msg.HasField("value") and not msg.HasField("second_field") and not msg.HasField("between") and not msg.HasField("reference_set"):
             return SummaryConstraint(
                 msg.first_field,
                 msg.op,
                 value=msg.value,
                 name=msg.name,
                 verbose=msg.verbose,
             )
-        elif msg.HasField("second_field") and not msg.HasField("value") and not msg.HasField("between"):
+        elif msg.HasField("second_field") and not msg.HasField("value") and not msg.HasField("between") and not msg.HasField("reference_set"):
             return SummaryConstraint(
                 msg.first_field,
                 msg.op,
                 second_field=msg.second_field,
                 name=msg.name,
                 verbose=msg.verbose,
             )
-        elif msg.HasField("between") and not msg.HasField("value") and not msg.HasField("second_field"):
+        elif msg.HasField("between") and not msg.HasField("value") and not msg.HasField("second_field") and not msg.HasField("reference_set"):
```

Comment:
This if condition could be turned into a function. Maybe sync with @pecop2 in this